### PR TITLE
Run code snippets in ReindexDocumentationIT

### LINKED ISSUE #32093: Run code snippets in ReindexDocumentationIT
Currently it seems we don't run any of the documentation code snippets in ReindexDocumentationIT.
Test methods don't start with the "test..." prefix, so we don't catch any errors in the documented code snippets other 
than compile errors. I think the snippets should be run like in other code documentation tests. However this might
require some additional test setup code. Also I think the api responses require some basic assertions. 
Otherwiew we miss errors like the one reported in https://github.com/elastic/elasticsearch/pull/32010.

### PR DESCRIPTION
**Pull request by vladimirdolzhenko**
Actually `ReindexDocumentationIT` has documentation code snippets but it does not run as no `test*`-methods there.

This PR enforces to run ReindexDocumentationIT code examples.

Closes #32093

### CODE REVIEW

**Review by cbuescher — APPROVED** [review]
Thanks for picking up that issue, I left a small comment but rest LGTM.

**Review by cbuescher — CHANGES_REQUESTED** [review]
Hm, the current CI failure seems to indicate that there might be problems with the testTask indexing. I didn't run into those when running the test locally with several repetitions though, can you check whats going on there?

File: modules/reindex/src/test/java/org/elasticsearch/client/documentation/ReindexDocumentationIT.java [hunk 1]
```diff
@@ -23,78 +23,132 @@
...
         // end::reindex1
     }
 
     @SuppressWarnings("unused")
-    public void updateByQuery() {
+    public void testUpdateByQuery() {
         Client client = client();
+        client.admin().indices().prepareCreate("foo").get();
+        client.admin().indices().prepareCreate("bar").get();
+        client.admin().indices().preparePutMapping(INDEX_NAME).setType("type").setSource("cat", "type=keyword").get();
```
[the remark this answers is not in the document]
Comment (vladimirdolzhenko) [reply modules/reindex/src/test/java/org/elasticsearch/client/documentation/ReindexDocumentationIT.java]: thanks for pointing to this 👍

**Comment by vladimirdolzhenko** [comment]
@cbismuth I guess you try to run it locally from IDE while it is quite reproducible locally with gradle.

TIL: if you want to use custom node plugins in integration test (I had to use to interfere into task execution process) you have to ignore to be run on external cluster (gradle does by default)
```
@Override
protected boolean ignoreExternalCluster() {
    return true;
}
```

otherwise `nodePlugins()` are completely ignored

**Review by cbuescher — APPROVED** [review]
Thanks for the explanation about the earlier CI problem. LGTM.

**Pull request merged**